app/src/webview/java/org/mozilla/focus/web/WebViewProvider.java

```diff
@@ -127,6 +127,7 @@ private static void configureDefaultSettings(Context context, WebSettings settin
     public static void applyAppSettings(Context context, WebSettings settings) {
         // We could consider calling setLoadsImagesAutomatically() here too (This will block images not loaded over the network too)
         settings.setBlockNetworkImage(Settings.getInstance(context).shouldBlockImages());
+        settings.setJavaScriptEnabled(!Settings.getInstance(context).shouldBlockJavaScript());
```

Comment:
This would only work for new WebView instances, right? So if I browse, go to settings, flips this toggle and go back to my browsing session then this WebView doesn't know about that, right? I think we had code somewhere that checks if a blocking setting has changed and updates the existing WebView - do we need to do that for JavaScript too?

Reply:
I think this is being handled already. `applyAppSettings()` in `WebViewProvider` is called by `onSharedPreferenceChanged()` in `SystemWebView`.

Reply:
Okay, perfect. :)